hazelcast-client-new/src/main/java/com/hazelcast/client/proxy/ClientReplicatedMapProxy.java

```diff
@@ -79,8 +82,14 @@
     protected static final String NULL_KEY_IS_NOT_ALLOWED = "Null key is not allowed!";
     protected static final String NULL_VALUE_IS_NOT_ALLOWED = "Null value is not allowed!";
 
+    private static final AtomicIntegerFieldUpdater<ClientReplicatedMapProxy> TARGET_PARTITION_ID_UPDATER =
+            AtomicIntegerFieldUpdater.newUpdater(ClientReplicatedMapProxy.class, "targetPartitionId");
+
     private volatile ClientHeapNearCache<Object> nearCache;
     private final AtomicBoolean nearCacheInitialized = new AtomicBoolean();
+    // all requests are forwarded to the same node via a random partition id
+    // do not use this field directly. use getOrInitTargetPartitionId() instead.
+    private volatile int targetPartitionId = Operation.GENERIC_PARTITION_ID;
```

Comment:
Doesn't it pick the partition of the name? I wanted to pick a random partition without triggering a partition table assignment on proxy creation so that there is no behavioural change while creating the proxy. If it is not a problem, I can get a random partition on proxy init.